app/controllers/admin/backups_controller.rb

```diff
@@ -119,7 +119,7 @@ def upload_backup_chunk
 
     return render status: 415, text: I18n.t("backup.backup_file_should_be_tar_gz") unless /\.(tar\.gz|t?gz)$/i =~ filename
     return render status: 415, text: I18n.t("backup.not_enough_space_on_disk")     unless has_enough_space_on_disk?(total_size)
-    return render status: 415, text: I18n.t("backup.invalid_filename") unless !!(/^[a-zA-Z0-9\.-_]+$/ =~ filename)
+    return render status: 415, text: I18n.t("backup.invalid_filename") unless !!(/^[a-zA-Z0-9\._-]+$/ =~ filename)
```

Comment:
Technically you don't have to when it ends with it, but you want it escaped anyway? As the test passes. I intentionally updated the test so it would fail if someone altered it and failed to escape the hyphen.